src/repositories/user-repository/UserRepository.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Inject } from '@graphql-modules/di';
+import { User } from './User';
+import { TableService, TableQuery } from 'azure-storage';
+
+export interface IUserRepository {
+  save(user: User): Promise<void>;
+  find(): Promise<User[]>;
+  tableName: string;
+}
+
+@Injectable()
+class UserRepository implements IUserRepository {
+  public tableName: string = 'userentity';
+
+  constructor(@Inject('TableService') private tableService: TableService) {
+    this.tableService.doesTableExist(this.tableName, (error, result) => {
+      if (!result.exists) {
+        this.tableService.createTable(this.tableName, (error, result) => {
+          console.log(error);
+          console.log(result);
+        });
+      }
+    });
+  }
+
+  public async save(user: User): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.tableService.insertEntity<User>(
+        this.tableName,
+        user,
+        (error) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve();
+          }
+        }
+      );
+    });
+  }
+
+  // @TODO: Add filters as a parameter, for now we are just returning
```

Comment:
Why don’t we just implement this now then?

Reply:
For that I need to investigate how to query the DB, which it will take me some time. I only have a couple of hours so I will do that next week, hence the TODO

Reply:
I also like smaller PRs 🤘